### ISSUE TITLE
Some Workflows Unavailable When Uploading Files

### ISSUE THREAD

**Issue by weiyiwang-dotcms** [dotCMS/core#27322]
### Problem Statement

When uploading a file via a content modal, only system workflows will show (assuming the authenticated user has access to system workflow). This means they cannot access the apply non-system workflows to uploaded files when uploading via a content modal.

### Steps to Reproduce

1. Create a content type that has a field for a file
2. Create a new workflow
3. Create a new folder
4. Give a limited user access to the new workflow, folder, and content type.
5. Log in as that user limited.
6. Create a new contentlet of the new content type.
7. For the file, click on browse, click on the new folder, and attempt to upload a file.
8. The selector for the workflow will only show the system workflow, but not the workflow steps for the new workflow.
9. Uploading the file via the site browser shows all of the possible workflow steps.

### Acceptance Criteria

Make the behavior consistent, where uploading files via a content modal has the same access to workflows as when uploading through the site browser.

### dotCMS Version

22.03.12

### Proposed Objective

Technical User Experience

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.zendesk.com/agent/tickets/114768
https://dotcms.slack.com/archives/G01N0AU1NUC/p1704907300092549

**Comment by gortiz-dotcms** [comment]
Hi @weiyiwang-dotcms we could not reproduce the error, here is a video in which I show all the steps already made:

- Limited access user
- Content type with file field, workflow and folder with the user permissions in each of them 
- File content type with access to new workflow.


Then trying to create a contentlet of the new content type and seeing that the new workflow is displayed when it shouldn't.

https://github.com/user-attachments/assets/c03c2f63-d856-444c-88bb-22742fc57140

 Can you please check and tell us if we are missing some steps to reproduce it?

**Comment by weiyiwang-dotcms** [comment]
I will take a look again tomorrow.

**Comment by weiyiwang-dotcms** [comment]
Hi, I figured out what the "issue" is. The availability of workflows when uploading a file as part of content creation is determined by what workflows are assigned to the fileAsset content type rather than the content type of the contentlet that contains the file.

**Comment by weiyiwang-dotcms** [comment]
Freshdesk Case
https://dotcms.freshdesk.com/a/tickets/24385

**Comment by weiyiwang-dotcms** [comment]
For this customer the issue is as below.

The default content type of the folder is Marketing Asset, but when you upload the file via the content screen, dotCMS adds it as a "File Asset".
![Screenshot 2024-09-12 at 12 32 00 PM](https://github.com/user-attachments/assets/4fb8ebd9-0d06-41ef-856b-3b008560f599)

Contents of "File Asset" type only allows for the System Workflow to run for it by default.
![Screenshot 2024-09-12 at 12 29 52 PM](https://github.com/user-attachments/assets/dd33fe21-685e-46a6-9dd6-2440e8232012)

The actions of the system workflow do not allow for the limited user to use them
![Screenshot 2024-09-12 at 12 34 35 PM](https://github.com/user-attachments/assets/500a084e-ca66-48ff-a4ac-4f3618b9d854)

In summary, I think this issue was all caused by dotCMS uploading the files to the folder as a "File Asset" content type, rather than as a "Marketing Asset" content type.

**Comment by erickgonzalez** [comment]
@weiyiwang-dotcms Yes, it seems that is the issue over here. Since the scope of this original issue was different, we will close it. Please create a new issue for this case so we can take a look.
